nms/app/packages/magmalte/app/state/feg/EquipmentState.js

```diff
@@ -74,6 +75,102 @@ export async function InitGatewayState(props: InitGatewayStateProps) {
   }
 }
 
+type GatewayStateProps = {
+  networkId: network_id,
+  fegGateways: {[gateway_id]: federation_gateway},
+  fegGatewaysHealthStatus: {[gateway_id]: FederationGatewayHealthStatus},
+  setFegGateways: ({[gateway_id]: federation_gateway}) => void,
+  setFegGatewaysHealthStatus: ({
+    [gateway_id]: FederationGatewayHealthStatus,
+  }) => void,
+  setActiveFegGatewayId: gateway_id => void,
+  key: gateway_id,
+  value?: mutable_federation_gateway,
+  newState?: {[gateway_id]: federation_gateway},
+  enqueueSnackbar: (
+    msg: string,
+    cfg: EnqueueSnackbarOptions,
+  ) => ?(string | number),
+};
+
+/**
+ * Adds, edits, or deletes a federation gateway or sets the gateway state to a new state. It
+ * then makes sure to sync the health status of the gateways and update the active gateway id
+ * in case it changed.
+ *
+ * @param {network_id} networkId Id of the federation network
+ * @param {{[gateway_id]: federation_gateway}} fegGateways Federation gateways of the network.
+ * @param {{[gateway_id]: FederationGatewayHealthStatus}} fegGatewaysHealthStatus Health status of the federation gateways.
+ * @param {({[string]: federation_gateway}) => void} setFegGateways Sets federation gateways.
+ * @param {({[string]: FederationGatewayHealthStatus}) => void} setFegGatewaysHealthStatus Sets federation gateways health status.
+ * @param {(gatewayId:gateway_id) => void} setActiveFegGatewayId Sets the active gateway id.
+ * @param {gateway_id} key Id of the gateway to be added, deleted or edited.
+ * @param {mutable_federation_gateway} value New Value for the gateway with the id: key.
+ * @param {[gateway_id]: federation_gateway} newState New State of the Federation Gateway.
+ * @param {(msg, cfg,) => ?(string | number),} enqueueSnackbar Snackbar to display error
```

Comment:
Specify the only `@param` as type `GatewayStateProps`, and add these helpful comments to the definition of `GatewayStateProps` above.